sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/CommonUtil.kt

```diff
@@ -0,0 +1,246 @@
+            }
+            sb.toString()
+        } catch (e: NoSuchAlgorithmException) {
+            null
+        }
+    }
+
+    /**
+     * Check the device to make sure it has the Google Play Services APK.
+     *
+     *
+     * If it doesn't, display a dialog that allows users to download the APK from the Google Play
+     * Store or enable it in the device's system settings.
+     *
+     * @param context non null activity, used to display update dialog if necessary.
+     * @return true if play services are available, false otherwise
+     */
+    fun checkPlayServices(context: Activity): Boolean {
+        val googleApi = GoogleApiAvailability.getInstance()
+        val result = googleApi.isGooglePlayServicesAvailable(context)
+        if (result != ConnectionResult.SUCCESS) {
+            if (googleApi.isUserResolvableError(result)) {
+                googleApi.getErrorDialog(context, result, PLAY_SERVICES_RESOLUTION_REQUEST)?.show()
+            }
+            return false
+        }
+        return true
+    }
+
+    /**
+     * Get the UTC date format from the simple date format, i.e. "yyyy-MM-dd'T'HH:mm:ssX"
+     *
+     * @return the date format
+     */
+    @SuppressLint("SimpleDateFormat")
+    fun getUTCDateFormat(): DateFormat {
+        val dateFormat: DateFormat = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssX")
+        } else {
+            SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss")
+        }
+        val timeZone = TimeZone.getTimeZone("UTC")
+        dateFormat.timeZone = timeZone
+        return dateFormat
+    }
+
+    /**
+     * Get the UTC date from a sample date string
+     *
+     * @param dateString the date string.
+     * @return the date object based on the date string
+     */
+    @SuppressLint("SimpleDateFormat")
+    fun getUTCDate(dateString: String?): Date? {
+        val dateFormat: DateFormat = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssX")
+        } else {
+            SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss")
+        }
+        val timeZone = TimeZone.getTimeZone("UTC")
+        dateFormat.timeZone = timeZone
+        return dateFormat.parse(dateString ?: "2021-12-01T00:00:00Z")
+    }
+
+    /**
+     * Get the resource id from drawable folder.
+     *
+     * @param context the application context.
+     * @param resourceName the resource name
+     * @return the drawable resource id
+     */
+    fun getDrawableResourceId(context: Context, resourceName: String?): Int {
+        return context.resources.getIdentifier(resourceName, "drawable", context.packageName)
+    }
+
+    /**
+     * Get the resource id from raw folder.
+     *
+     * @param context the application context.
+     * @param resourceName the resource name
+     * @return the raw resource id
+     */
+    fun getRawResourceId(context: Context, resourceName: String?): Int {
```

Comment:
Can we make this as extension of `Context` class?

Reply:
Done